core/dynamo.c

```diff
@@ -978,7 +984,22 @@ dynamo_shared_exit(thread_record_t *toexit /* must ==cur thread for Linux */
      * client trying to use api routines that depend on fragment state.
      */
     instrument_exit();
-#endif
+# ifdef CLIENT_SIDELINE
+    /* We only need do a second synch-all if there are sideline client threads. */
```

Comment:
You mean that even w/o the lock, a new thread will pause.  Sure, so no downside to adding a num threads check.